tools/cmake/python_packagecheck.py

```diff
@@ -0,0 +1,16 @@
+"""
+mbed SDK
+Copyright (c) 2020 ARM Limited
+SPDX-License-Identifier: Apache-2.0
+"""
+
+#file which is invoked by the cmake build system to check if all necessary python packages are installed.
+
+import sys
+
+try:
+    __import__(sys.argv[1])
+except ImportError:
+    exit(1)
+
+exit(0)
```

Comment:
I would rewrite it as follows:

```suggestion

import argparse
import sys
from enum import Enum


class ReturnCode(Enum):
    """Return codes."""

    SUCCESS = 0
    ERROR = 1
    INVALID_OPTIONS = 2


def parse_args():
    """Parse the command line args."""
    parser = argparse.ArgumentParser(
        description="Check if a Python package/module has been installed."
    )

    parser.add_argument(
        "python-package",
        type=str,
        help="the Python package/module to check.",
    )

    args = parser.parse_args()


def run_python_packagecheck():
    """Application main algorithm."""
    args = parse_args()
    __import__(args.python_package)


def _main():
    """Run python-packagecheck."""
    try:
        run_python_packagecheck()
    except Exception as error:
        print(error)
        return ReturnCode.ERROR.value
    else:
        return ReturnCode.SUCCESS.value


if __name__ == "__main__":
    sys.exit(_main())
```

Reply:
Okaaay... this feels like it's way overcomplicating things.  I realized that this script can just be removed entirely.